### PR TITLE
feat: emit through forwarder

### PR DESCRIPTION
**Pull request by GalloDaSballo**

### CODE REVIEW

File: contracts/VotiumBribesProcessor.sol [hunk 1]
```diff
@@ -191,12 +187,20 @@ contract VotiumBribesProcessor is CowSwapSeller {
...
             // If we don't swap
+
+            // Take the fee
             BVE_CVX.depositFor(TREASURY, ops_fee);
-            BVE_CVX.depositFor(BADGER_TREE, toEmit);
+
+            // Deposit and emit rest
+            uint256 initialBveCVXBalance = BVE_CVX.balanceOf((address(this)));
+            BVE_CVX.deposit(toEmit);
+            uint256 afterBveCVXBalance = BVE_CVX.balanceOf((address(this)));
```
Comment (mitche50) [inline contracts/VotiumBribesProcessor.sol:197]: You set this var here and never use it

File: contracts/VotiumBribesProcessor.sol [hunk 2]
```diff
@@ -210,12 +214,15 @@ contract VotiumBribesProcessor is CowSwapSeller {
...
             toEmit = totalBveCVX - ops_fee;
 
+            // Take fee
             IERC20(address(BVE_CVX)).safeTransfer(TREASURY, ops_fee);
-            IERC20(address(BVE_CVX)).safeTransfer(BADGER_TREE, toEmit);
         }
 
+        // Emit token
+        IERC20(address(BVE_CVX)).safeApprove(address(HARVEST_FORWARDER), toEmit);
+        HARVEST_FORWARDER.distribute(address(BVE_CVX), toEmit, address(BVE_CVX));
```
Comment (mitche50) [inline contracts/VotiumBribesProcessor.sol:223]: A good sanity check before distribution is to assert `HARVEST_FORWARDER.badger_tree == BADGER_TREE`

File: contracts/VotiumBribesProcessor.sol [hunk 3]
```diff
@@ -228,9 +235,8 @@ contract VotiumBribesProcessor is CowSwapSeller {
         uint256 toEmit = BADGER.balanceOf(address(this));
         require(toEmit > 0);
 
-        BADGER.safeTransfer(BADGER_TREE, toEmit);
-
-        emit TreeDistribution(address(BADGER), toEmit, block.number, block.timestamp);
+        BADGER.safeApprove(address(HARVEST_FORWARDER), toEmit);
+        HARVEST_FORWARDER.distribute(address(BADGER), toEmit, address(BVE_CVX));
```
Comment (mitche50) [inline contracts/VotiumBribesProcessor.sol:239]: There should be two distributions here based on a percent split we can modify:
- bveCVX distribution
- bveCVX LP distribution

Details of the implementation are here:
https://github.com/Badger-Finance/badger-multisig/blob/56bb76914b9e665997702fe2e82e08a4ffe1599f/scripts/emissions/bribes_emissions.py#L18-L23

File: interfaces/badger/IHarvestForwarder.sol [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+pragma solidity >=0.6.11;
+pragma experimental ABIEncoderV2;
+
+interface IHarvestForwarder {
+  function distribute(
+    address token,
+    uint256 amount,
+    address beneficiary
+  ) external;
```
Comment (mitche50) [inline interfaces/badger/IHarvestForwarder.sol:11]: Add below for sanity check
```
function badger_tree() external view returns(address);
```

**Review by mitche50 — COMMENTED** [review]
General comment, maybe we should still be emitting bribe processor specific events so we can set up monitoring / automated analysis tools

**Comment by GalloDaSballo** [comment]
Closes #9

**Pull request merged**